lib/query_relation/array_based_collection.rb

```diff
@@ -0,0 +1,14 @@
+class QueryRelation
+  module ArrayBasedCollection
+    # Default filter for an array based +collection+ based on various hash
+    # based +options+ that are used by QueryRelation.
+    #
+    def filter_array_collection(collection, options)
+      collection = collection.drop(options[:offset]) if options[:offset]
+      collection = collection.take(options[:limit]) if options[:limit]
+      collection = collection.select{ |hash| hash.slice(*options[:where].keys) == options[:where] } if options[:where]
```

Comment:
think this needs to be a little more complicated.

think we need to each through the `options[:where]` values.

I found the different values for the `where` needed to use different types of equality.

see https://github.com/kbrock/activerecord-hash_options/blob/master/lib/active_record/hash_options.rb#L49-L80

Reply:
Well, remember, this is just meant to be a basic default implementation.

If we want to get fancy, we should probably just incorporate your library instead of reinventing the wheel.